handler/interceptors.go

```diff
@@ -235,10 +244,72 @@ func GrpcPaymentValidationInterceptor(defaultPaymentHandler PaymentHandler, paym
 }
 
 type paymentValidationInterceptor struct {
+	serviceMetadata       *blockchain.ServiceMetadata
 	defaultPaymentHandler PaymentHandler
 	paymentHandlers       map[string]PaymentHandler
 }
 
+func (interceptor *paymentValidationInterceptor) checkForDynamicPricing(srv interface{}, ss grpc.ServerStream,
```

Comment:
My original intention was isolating pricing related logic in `pricing` module. In particular such implementation will allow get price dynamically only when it is needed (i.e. don't do this when `free-call` or `authorized-user` payment strategy is applied).

If I would implement this I would do the following:
- move checking if dynamic pricing is applicable to `PricingStrategy.determincePricingApplicable` method
- add new `DYNAMIC_PRICING` into `PriceType`
- add new `dynamic_method_pricing.go` and move getting price related code into it (similar to `fixed_method_pricing.go`
- to make above possible probably we need to modify `handler.GrpcStreamContext` and add additional information into it if needed
- `ServiceMetadata` should be part of the `PricingStrategy` I believe, but I am not sure how to implement it better

Reply:
made changes , if it looks fine , will check in the test cases as well